sale_order_warehouse_from_delivery_carrier/models/sale_order.py

```diff
@@ -0,0 +1,21 @@
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends(
+        "user_id", "company_id", "state", "carrier_id", "carrier_id.so_warehouse_id"
```

Comment:
This wasn't solved. I don't see a reason for recomputation on state change. And you don't want to recompute all your SO if the config on the carrier is changed
```suggestion
        "user_id", "company_id", "carrier_id"
```

Reply:
Agree, didn't see